src/environmentd/tests/sql.rs

```diff
@@ -1341,9 +1336,28 @@ fn test_utilization_hold() {
                 .execute(&format!("SET cluster={cluster}"), &[])
                 .unwrap();
 
-            let row = client.query_one(explain_q, &[]).unwrap();
-            let explain: String = row.get(0);
-            let explain: TimestampExplanation<Timestamp> = serde_json::from_str(&explain).unwrap();
+            // Hack: we think there might be an issue where sinces
+            // can briefly be zero on startup, which breaks our logic here.
+            //
+            // So just spin until it's not zero.
+            // TODO[btv] - Get rid of this loop if that bug is ever fixed
+            let mut n_sleeps = 0;
+            let explain = loop {
+                let row = client.query_one(explain_q, &[]).unwrap();
+                let explain: String = row.get(0);
+                let explain: TimestampExplanation<Timestamp> =
+                    serde_json::from_str(&explain).unwrap();
+                if explain.determination.since.clone().into_option() == Some(Timestamp::MIN) {
+                    n_sleeps += 1;
+                    if n_sleeps > 10 {
+                        panic!("Since never became non-zero");
+                    }
+                    println!("Since is 0, sleeping for 1 second");
+                    std::thread::sleep(Duration::from_secs(1));
+                } else {
+                    break explain;
+                }
+            };
```

Comment:
We should probably use `mz_ore::retry` to do this. Grep for `Retry::default()` in this file for examples.